### ISSUE TITLE
Plugin does not find/add all devices

### ISSUE THREAD

**Issue by faraz** [tuya/tuya-homebridge#40]
A total of 35 devices are registered on the smartlife app and visible on the project's device list. However, when the plugin runs, it is unable to discover and add all devices - I counted it can only add 20 devices. I tried to clear homebridge's device cache but no luck.

In the logs, I noticed that the below call is only returning 20 `device_ids`
```
TuyaOpenAPI request: method = get, endpoint = https://openapi.tuyaus.com, path = /v1.0/devices/functions,
```

```
TuyaOpenAPI response: {"result":{"devices":[{"active_time":1592788990,"biz_type":18,"category":"dj","create_time":1591563960,"icon":"smart/product_icon/dj.png","id":"04331260a4cf12ac430f","ip":"REMOVED","lat":"REMOVED","local_key":"REMOVED","lon":"-80.81315117760964","model":"","name":"Loft Rail Light 5","online":true,"owner_id":"18281334","product_id":"QP2puy93gbcNET5G","product_name":"RGBCW","status":[{"code":"switch_led","value":false},{"code":"work_mode","value":"colour"},{"code":"bright_value","value":255},{"code":"temp_value","value":127},{"code":"colour_data","value":"{\"h\":21.0,\"s\":255.0,\"v\":43.1}"},{"code":"scene_data","value":"{\"h\":16.2,\"s\":255.0,\"v\":207.0}"},{"code":"flash_scene_1","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":120.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_2","value":"{\"bright\":255,\"frequency\":128,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":120.0,\"s\":255.0,\"v\":255.0},{\"h\":240.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0}],\"temperature\":255}"},{"code":"flash_scene_3","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_4","value":"{\"bright\":255,\"frequency\":5,\"hsv\":[{\"h\":176.9,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284821,"uuid":"04331260a4cf12ac430f"},{"active_time":1607643364,"biz_type":18,"category":"tgq","create_time":1607643364,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"8624750310521cd4434b","ip":"REMOVED","lat":"REMOVED","local_key":"47812b7438adb5d3","lon":"-80.81315117760964","name":"Bar Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":650},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"8624750310521cd4434b"},{"active_time":1610154862,"biz_type":18,"category":"tgq","create_time":1610154862,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"4736863710521cfb42b9","ip":"REMOVED","lat":"REMOVED","local_key":"REMOVED","lon":"-80.81315117760964","name":"Upstairs Bathroom Hallway Light","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"4736863710521cfb42b9"},{"active_time":1610654649,"biz_type":18,"category":"kg","create_time":1610654649,"icon":"smart/icon/ay14906729310721fNjn/1568862445415ffab0175.jpg","id":"017743508caab5f290c0","ip":"REMOVED","lat":"REMOVED","local_key":"19424cd1740f1478","lon":"-80.81315117760964","name":"Guest Bedroom Closet Light","online":true,"owner_id":"18281334","product_id":"pigcdkf9bdubtskr","product_name":"SS01S Smart Light Switch","status":[{"code":"switch_1","value":false},{"code":"countdown_1","value":0}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"017743508caab5f290c0"},{"active_time":1610230276,"biz_type":18,"category":"tgq","create_time":1610230276,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"473686375002912791ed","ip":"REMOVED","lat":"REMOVED","local_key":"acbc0871ebb24a2e","lon":"-80.81315117760964","name":"Basement Playroom Light","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"473686375002912791ed"},{"active_time":1609342225,"biz_type":18,"category":"tgq","create_time":1609342225,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"4736863710521cd65f5b","ip":"REMOVED","lat":"REMOVED","local_key":"633fdca06fedd23a","lon":"-80.81315117760964","name":"Butler's Pantry Light","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":800},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"4736863710521cd65f5b"},{"active_time":1610068266,"biz_type":18,"category":"tgq","create_time":1610068266,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"40855140f4cfa2306654","ip":"REMOVED","lat":"REMOVED","local_key":"44ef6d2b08c1cd6d","lon":"-80.81315117760964","name":"Foyer Chandelier","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"40855140f4cfa2306654"},{"active_time":1623623896,"biz_type":18,"category":"tgq","create_time":1623623896,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"40374727c82b96ec9c31","ip":"REMOVED","lat":"REMOVED","local_key":"d95a400d2941a4c1","lon":"-80.81315117760964","name":"Nadia Room Ceiling Light","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"40374727c82b96ec9c31"},{"active_time":1592788729,"biz_type":18,"category":"dj","create_time":1591563657,"icon":"smart/product_icon/dj.png","id":"04331260a4cf12ac9324","ip":"REMOVED","lat":"REMOVED","local_key":"23cec50b5dc7b6de","lon":"-80.81315117760964","model":"","name":"Loft Rail Light 4","online":true,"owner_id":"18281334","product_id":"QP2puy93gbcNET5G","product_name":"RGBCW","status":[{"code":"switch_led","value":false},{"code":"work_mode","value":"colour"},{"code":"bright_value","value":255},{"code":"temp_value","value":127},{"code":"colour_data","value":"{\"h\":21.0,\"s\":255.0,\"v\":43.1}"},{"code":"scene_data","value":"{\"h\":16.2,\"s\":255.0,\"v\":207.0}"},{"code":"flash_scene_1","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":120.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_2","value":"{\"bright\":255,\"frequency\":128,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":120.0,\"s\":255.0,\"v\":255.0},{\"h\":240.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0}],\"temperature\":255}"},{"code":"flash_scene_3","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_4","value":"{\"bright\":255,\"frequency\":5,\"hsv\":[{\"h\":176.9,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"04331260a4cf12ac9324"},{"active_time":1610155160,"biz_type":18,"category":"tgq","create_time":1610155160,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"47368637500291277afb","ip":"REMOVED","lat":"REMOVED","local_key":"7a7bed9b9111dfe6","lon":"-80.81315117760964","name":"Stairs Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":200},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"47368637500291277afb"},{"active_time":1592788533,"biz_type":18,"category":"dj","create_time":1591562958,"icon":"smart/product_icon/dj.png","id":"0433126084f3ebd19349","ip":"REMOVED","lat":"REMOVED","local_key":"311e6df62e40b2e3","lon":"-80.81315117760964","model":"","name":"Loft Rail Light 1","online":true,"owner_id":"18281334","product_id":"QP2puy93gbcNET5G","product_name":"RGBCW","status":[{"code":"switch_led","value":false},{"code":"work_mode","value":"colour"},{"code":"bright_value","value":255},{"code":"temp_value","value":127},{"code":"colour_data","value":"{\"h\":21.0,\"s\":255.0,\"v\":43.1}"},{"code":"scene_data","value":"{\"h\":16.2,\"s\":255.0,\"v\":207.0}"},{"code":"flash_scene_1","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":120.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_2","value":"{\"bright\":255,\"frequency\":128,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":120.0,\"s\":255.0,\"v\":255.0},{\"h\":240.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0}],\"temperature\":255}"},{"code":"flash_scene_3","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_4","value":"{\"bright\":255,\"frequency\":5,\"hsv\":[{\"h\":176.9,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"0433126084f3ebd19349"},{"active_time":1610155024,"biz_type":18,"category":"tgq","create_time":1610155024,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"4736863710521cfb82a7","ip":"REMOVED","lat":"REMOVED","local_key":"a483f046176d15ca","lon":"-80.81315117760964","name":"Loft Hallway Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":true},{"code":"bright_value_1","value":500},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"4736863710521cfb82a7"},{"active_time":1610989480,"biz_type":18,"category":"tgq","create_time":1610989480,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"403747275002910a023b","ip":"REMOVED","lat":"REMOVED","local_key":"a9607987ac1b8e48","lon":"-80.81315117760964","name":"Breakfast Area Chandelier","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"403747275002910a023b"},{"active_time":1610989726,"biz_type":18,"category":"tgq","create_time":1610989726,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"85581320c82b9658ff47","ip":"REMOVED","lat":"REMOVED","local_key":"4935b785d2cab747","lon":"-80.81315117760964","name":"Master Bedroom Door Light","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":650},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"85581320c82b9658ff47"},{"active_time":1592789102,"biz_type":18,"category":"dj","create_time":1591563633,"icon":"smart/product_icon/dj.png","id":"04331260a4cf12ad25ba","ip":"REMOVED","lat":"REMOVED","local_key":"3c22c4e3a0ac5140","lon":"-80.81315117760964","model":"","name":"Loft Rail Light 2","online":true,"owner_id":"18281334","product_id":"QP2puy93gbcNET5G","product_name":"RGBCW","status":[{"code":"switch_led","value":false},{"code":"work_mode","value":"colour"},{"code":"bright_value","value":255},{"code":"temp_value","value":127},{"code":"colour_data","value":"{\"h\":21.0,\"s\":255.0,\"v\":43.1}"},{"code":"scene_data","value":"{\"h\":16.2,\"s\":255.0,\"v\":207.0}"},{"code":"flash_scene_1","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":120.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_2","value":"{\"bright\":255,\"frequency\":128,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":120.0,\"s\":255.0,\"v\":255.0},{\"h\":240.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0}],\"temperature\":255}"},{"code":"flash_scene_3","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_4","value":"{\"bright\":255,\"frequency\":5,\"hsv\":[{\"h\":176.9,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"04331260a4cf12ad25ba"},{"active_time":1623698691,"biz_type":18,"category":"tgq","create_time":1610989549,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"2188570270039fcfd566","ip":"REMOVED","lat":"REMOVED","local_key":"e5ca0b62e253ba15","lon":"-80.81315117760964","name":"Foyer Hallway Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":350},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"2188570270039fcfd566"},{"active_time":1608755679,"biz_type":18,"category":"tgq","create_time":1608755618,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"82680347f4cfa23a09e0","ip":"REMOVED","lat":"REMOVED","local_key":"86569b758646c3d5","lon":"-80.81315117760964","name":"Master Bathroom Shower Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":1000},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"82680347f4cfa23a09e0"},{"active_time":1608755472,"biz_type":18,"category":"tgq","create_time":1608755472,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"4736863750029127f56b","ip":"REMOVED","lat":"REMOVED","local_key":"f13471665f5fcd53","lon":"-80.81315117760964","name":"Master Bedroom Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":650},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"incandescent"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"4736863750029127f56b"},{"active_time":1609265728,"biz_type":18,"category":"tgq","create_time":1609265728,"icon":"smart/icon/ay1517349698958w8yNe/4a306619b2cf81cb894bde78821e0657.png","id":"4736863750029127ba9c","ip":"REMOVED","lat":"REMOVED","local_key":"f2f7b3058d4bca98","lon":"-80.81315117760964","name":"Kitchen Island Lights","online":true,"owner_id":"18281334","product_id":"tebc75erlgslgpn5","product_name":"Dimmer  Switch","status":[{"code":"switch_led_1","value":false},{"code":"bright_value_1","value":200},{"code":"brightness_min_1","value":10},{"code":"led_type_1","value":"LED"},{"code":"work_mode","value":"white"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"4736863750029127ba9c"},{"active_time":1592788887,"biz_type":18,"category":"dj","create_time":1591563637,"icon":"smart/product_icon/dj.png","id":"83608330cc50e3cb33e2","ip":"REMOVED","lat":"REMOVED","local_key":"3ac0202faa7329d5","lon":"-80.81315117760964","model":"","name":"Loft Rail Light 3","online":true,"owner_id":"18281334","product_id":"QP2puy93gbcNET5G","product_name":"RGBCW","status":[{"code":"switch_led","value":false},{"code":"work_mode","value":"colour"},{"code":"bright_value","value":255},{"code":"temp_value","value":127},{"code":"colour_data","value":"{\"h\":21.0,\"s\":255.0,\"v\":43.1}"},{"code":"scene_data","value":"{\"h\":16.2,\"s\":255.0,\"v\":207.0}"},{"code":"flash_scene_1","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":120.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_2","value":"{\"bright\":255,\"frequency\":128,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":120.0,\"s\":255.0,\"v\":255.0},{\"h\":240.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0},{\"h\":0.0,\"s\":0.0,\"v\":0.0}],\"temperature\":255}"},{"code":"flash_scene_3","value":"{\"bright\":255,\"frequency\":80,\"hsv\":[{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"},{"code":"flash_scene_4","value":"{\"bright\":255,\"frequency\":5,\"hsv\":[{\"h\":176.9,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":258.1,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0},{\"h\":0.0,\"s\":255.0,\"v\":255.0}],\"temperature\":255}"}],"sub":false,"time_zone":"-05:00","uid":"az1591562844895JgQkL","update_time":1624284820,"uuid":"83608330cc50e3cb33e2"}],"has_more":true,"last_row_key":"4F492D1176ED54C96A4DBE8C55B77882","total":35},"success":true,"t":1624284887434} path = /v1.0/iot-01/associated-users/devices
```

**Comment by Fastasabadger** [comment]
I had this issue as well.  The only way I managed to get around it was to re-add the missing devices to the Tuya smartphone App and then restart the plugin.  In my case I had about 7 outlets which did not show up on the Homebridge.

Try re-adding one of your offending devices, if that makes it appear then it's the same issue I had which is definitely another bug.

**Comment by faraz** [comment]
> I had this issue as well. The only way I managed to get around it was to re-add the missing devices to the Tuya smartphone App and then restart the plugin. In my case I had about 7 outlets which did not show up on the Homebridge.
> 
> Try re-adding one of your offending devices, if that makes it appear then it's the same issue I had which is definitely another bug.

That's interesting.. the `product_id` for the missing devices is the same as the ones it did manage to find. Must be a bug on the API side.

**Comment by Fastasabadger** [comment]
Agreed, I found some outlets with the same manufacturer and product_id appeared some didn't and then some from another manufacturer appeared some didn't.  There didn't seem to be any corrolation between appearing or not, just luck of the draw!

Highly annoying as I had to spend ages going through and working out which devices I was missing from Homekit but appeared in the Tuya App.

**Comment by faraz** [comment]
@Fastasabadger appreciate the input. I suppose the last resort is to re-pair the missing devices with SmartLife again... I am hoping @Hanh94 or someone else from Tuya may have an official fix

**Comment by Hanh94** [comment]
Hi @faraz @Fastasabadger ,Thank you for your feedback.
We will check this issue and reply as soon as possible.

**Comment by Hanh94** [comment]
Hi @faraz @Fastasabadger ,We have fixed this issue in the new version v1.2.0, please update and try again.

**Comment by faraz** [comment]
@Hanh94 I can confirm that https://github.com/tuya/tuya-homebridge/commit/36afdddebcac4f23abf09f48e1838f8c5fa0c083  / v1.2.0 fixes this issue. All 35 of my devices now get picked up.
